python/lsst/pipe/analysis/colorAnalysis.py

```diff
@@ -363,281 +560,1040 @@ def plotGalaxyColors(self, catalogs, filenamer, dataId):
             catalog.writeFits("gi.fits")
             shortName = "galaxy-TEST"
             self.log.info("shortName = {:s}".format(shortName))
-            self.AnalysisClass(catalog, GalaxyColor("modelfit_CModel_flux", "slot_CalibFlux_flux", "g_", "i_"),
+            self.AnalysisClass(catalog,
+                               GalaxyColor("modelfit_CModel_flux", "slot_CalibFlux_flux", "g_", "i_"),
                                "(g-i)_cmodel - (g-i)_CalibFlux", shortName, self.config.analysis,
                                flags=["modelfit_CModel_flag", "slot_CalibFlux_flag"], prefix="i_",
                                labeller=OverlapsStarGalaxyLabeller("g_", "i_"),
                                qMin=-0.5, qMax=0.5,).plotAll(dataId, filenamer, self.log)
 
-    def plotStarColors(self, catalog, filenamer, labeller, dataId, butler=None, camera=None, tractInfo=None,
-                       patchList=None, hscRun=None):
+    def plotStarColors(self, principleColCats, byFilterCats, filenamer, labeller, dataId, butler=None,
+                       camera=None, tractInfo=None, patchList=None, hscRun=None):
+        mags = {ff: -2.5*np.log10(byFilterCats[ff]["base_PsfFlux_flux"]) for ff in byFilterCats}
+        color = lambda c1, c2: (mags[c1] - mags[c2])
         unitStr = "mmag" if self.config.toMilli else "mag"
         for col, transform in self.config.transforms.items():
-            if not transform.plot or col not in catalog.schema:
+            if not transform.plot or col not in principleColCats.schema:
                 continue
+            if self.config.transforms == ivezicTransformsHSC:
+                if col == "wPerp" or col == "xPerp":
+                    colStr1, colStr2, colStr3 = "HSC-G", "HSC-R", "HSC-I"
+                    filterStrList = ["g", "r", "i", ""]
+                elif col == "yPerp":
+                    colStr1, colStr2, colStr3 = "HSC-R", "HSC-I", "HSC-Z"
+                    filterStrList = ["r", "i", "z", ""]
+                else:
+                    raise RuntimeError("Unknown transformation name: {:s}".format(col))
```

Comment:
Do you prefer to raise if someone put an unknown transform in the config, or just skip that one and try the next?